### PR TITLE
Fix commands flags

### LINKED ISSUE #130: `evaluate` currently doesn't work
`AttributeError: 'Namespace' object has no attribute 'reproducible'`

Look at fix on `transfer-learning` branch.

### LINKED ISSUE #131: Default command in readme has wrong syntax
The example command to run training is missing a backslash and contains the seed twice leading to missed flags, it should probably be:

```
icu-benchmarks train \
    -d demo_data/mortality24/mimic_demo \
    -n mimic_demo \
    -t BinaryClassification \
    -tn Mortality24 \
    -m LGBMClassifier \
    -hp LGBMClassifier.min_child_samples=10 \
    --generate_cache \
    --load_cache \
    --seed 2222 \
    -l ../yaib_logs/ \
    --tune
```

### PR DESCRIPTION
**Pull request by HendrikSchmidt**
Removes train and evaluate commands and moves to a flag only system, also supporting finetuning (more work here is needed for full support, check branch https://github.com/rvandewater/YAIB/tree/transfer-learning ).

fixes #130 
fixes #131

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -240,10 +239,13 @@ icu-benchmarks evaluate \
     --load_cache \
     -s 2222 \
     -l ../yaib_logs \
+    --eval \
```
Comment (rvandewater) [inline README.md:242]: put eval as the first flag for readability?

Comment (HendrikSchmidt) [reply README.md:242]: yes, had a similar thought. wanted to leave it close to the flags that depend on it, but also agree with your sentiment. will change.

**Comment by rvandewater** [comment]
When executing the following command:

```
-d
demo_data/mortality24/eicu_demo
-n
eicu_demo
-t
BinaryClassification
-tn
Mortality24
-m
GRU
--generate_cache
--load_cache
-s
2222
-l
../yaib_logs
--eval
-sn
eicu
--source-dir
C:\Users\Robin\Downloads\gru_creat\eicu\repetition_0\fold_0\
```
I am getting this error: 
```
2023-08-15 15:55:15 - INFO - root : Using 4 workers for data loading.
Traceback (most recent call last):
  File "C:\Users\Robin\Documents\Git\YAIB\icu_benchmarks\run.py", line 178, in <module>
    main()
  File "C:\Users\Robin\Documents\Git\YAIB\icu_benchmarks\run.py", line 150, in main
    execute_repeated_cv(
  File "C:\Users\Robin\miniconda\envs\yaib_15\lib\site-packages\gin\config.py", line 1605, in gin_wrapper
    utils.augment_exception_message_and_reraise(e, err_str)
  File "C:\Users\Robin\miniconda\envs\yaib_15\lib\site-packages\gin\utils.py", line 41, in augment_exception_message_and_reraise
    raise proxy.with_traceback(exception.__traceback__) from None
  File "C:\Users\Robin\miniconda\envs\yaib_15\lib\site-packages\gin\config.py", line 1582, in gin_wrapper
    return fn(*new_args, **new_kwargs)
  File "C:\Users\Robin\Documents\Git\YAIB\icu_benchmarks\cross_validation.py", line 94, in execute_repeated_cv
    agg_loss += train_common(
  File "C:\Users\Robin\miniconda\envs\yaib_15\lib\site-packages\gin\config.py", line 1605, in gin_wrapper
    utils.augment_exception_message_and_reraise(e, err_str)
  File "C:\Users\Robin\miniconda\envs\yaib_15\lib\site-packages\gin\utils.py", line 41, in augment_exception_message_and_reraise
    raise proxy.with_traceback(exception.__traceback__) from None
  File "C:\Users\Robin\miniconda\envs\yaib_15\lib\site-packages\gin\config.py", line 1582, in gin_wrapper
    return fn(*new_args, **new_kwargs)
  File "C:\Users\Robin\Documents\Git\YAIB\icu_benchmarks\models\train.py", line 121, in train_common
    model = model.load_state_dict(checkpoint)
  File "C:\Users\Robin\miniconda\envs\yaib_15\lib\site-packages\torch\nn\modules\module.py", line 2041, in load_state_dict
    raise RuntimeError('Error(s) in loading state_dict for {}:\n\t{}'.format(
RuntimeError: Error(s) in loading state_dict for GRUNet:
	Missing key(s) in state_dict: "rnn.weight_ih_l0", "rnn.weight_hh_l0", "rnn.bias_ih_l0", "rnn.bias_hh_l0", "rnn.weight_ih_l1", "rnn.weight_hh_l1", "rnn.bias_ih_l1", "rnn.bias_hh_l1", "rnn.weight_ih_l2", "rnn.weight_hh_l2", "rnn.bias_ih_l2", "rnn.bias_hh_l2", "logit.weight", "logit.bias". 
	Unexpected key(s) in state_dict: "epoch", "global_step", "pytorch-lightning_version", "state_dict", "loops", "callbacks", "optimizer_states", "lr_schedulers", "hparams_name", "hyper_parameters", "class", "trained_columns". 
  In call to configurable 'train_common' (<function train_common at 0x0000015C1E0632E0>)
  In call to configurable 'execute_repeated_cv' (<function execute_repeated_cv at 0x0000015C1BF2C040>)
```

**Comment by rvandewater** [comment]
> When executing the following command:
> 
> ```
> -d
> demo_data/mortality24/eicu_demo
> -n
> eicu_demo
> -t
> BinaryClassification
> -tn
> Mortality24
> -m
> GRU
> --generate_cache
> --load_cache
> -s
> 2222
> -l
> ../yaib_logs
> --eval
> -sn
> eicu
> --source-dir
> C:\Users\Robin\Downloads\gru_creat\eicu\repetition_0\fold_0\
> ```
> 
> I am getting this error:
> 
> ```
> 2023-08-15 15:55:15 - INFO - root : Using 4 workers for data loading.
> Traceback (most recent call last):
>   File "C:\Users\Robin\Documents\Git\YAIB\icu_benchmarks\run.py", line 178, in <module>
>     main()
>   File "C:\Users\Robin\Documents\Git\YAIB\icu_benchmarks\run.py", line 150, in main
>     execute_repeated_cv(
>   File "C:\Users\Robin\miniconda\envs\yaib_15\lib\site-packages\gin\config.py", line 1605, in gin_wrapper
>     utils.augment_exception_message_and_reraise(e, err_str)
>   File "C:\Users\Robin\miniconda\envs\yaib_15\lib\site-packages\gin\utils.py", line 41, in augment_exception_message_and_reraise
>     raise proxy.with_traceback(exception.__traceback__) from None
>   File "C:\Users\Robin\miniconda\envs\yaib_15\lib\site-packages\gin\config.py", line 1582, in gin_wrapper
>     return fn(*new_args, **new_kwargs)
>   File "C:\Users\Robin\Documents\Git\YAIB\icu_benchmarks\cross_validation.py", line 94, in execute_repeated_cv
>     agg_loss += train_common(
>   File "C:\Users\Robin\miniconda\envs\yaib_15\lib\site-packages\gin\config.py", line 1605, in gin_wrapper
>     utils.augment_exception_message_and_reraise(e, err_str)
>   File "C:\Users\Robin\miniconda\envs\yaib_15\lib\site-packages\gin\utils.py", line 41, in augment_exception_message_and_reraise
>     raise proxy.with_traceback(exception.__traceback__) from None
>   File "C:\Users\Robin\miniconda\envs\yaib_15\lib\site-packages\gin\config.py", line 1582, in gin_wrapper
>     return fn(*new_args, **new_kwargs)
>   File "C:\Users\Robin\Documents\Git\YAIB\icu_benchmarks\models\train.py", line 121, in train_common
>     model = model.load_state_dict(checkpoint)
>   File "C:\Users\Robin\miniconda\envs\yaib_15\lib\site-packages\torch\nn\modules\module.py", line 2041, in load_state_dict
>     raise RuntimeError('Error(s) in loading state_dict for {}:\n\t{}'.format(
> RuntimeError: Error(s) in loading state_dict for GRUNet:
> 	Missing key(s) in state_dict: "rnn.weight_ih_l0", "rnn.weight_hh_l0", "rnn.bias_ih_l0", "rnn.bias_hh_l0", "rnn.weight_ih_l1", "rnn.weight_hh_l1", "rnn.bias_ih_l1", "rnn.bias_hh_l1", "rnn.weight_ih_l2", "rnn.weight_hh_l2", "rnn.bias_ih_l2", "rnn.bias_hh_l2", "logit.weight", "logit.bias". 
> 	Unexpected key(s) in state_dict: "epoch", "global_step", "pytorch-lightning_version", "state_dict", "loops", "callbacks", "optimizer_states", "lr_schedulers", "hparams_name", "hyper_parameters", "class", "trained_columns". 
>   In call to configurable 'train_common' (<function train_common at 0x0000015C1E0632E0>)
>   In call to configurable 'execute_repeated_cv' (<function execute_repeated_cv at 0x0000015C1BF2C040>)
> ```

I am using a different task as a target I realized, so this is probably not properly supported which is fine atm.

**Pull request merged**